conda_lock/conda_lock.py

```diff
@@ -378,7 +378,8 @@ def make_lock_files(  # noqa: C901
 
         if platforms_to_lock:
             print(f"Locking dependencies for {platforms_to_lock}...", file=sys.stderr)
-            lock_content = lock_content | create_lockfile_from_spec(
```

Comment:
We believe the original intent of this merge was to persist platforms that were not requested for update (based on [this](https://github.com/conda/conda-lock/commit/6ea3f1cb212ed77216ff27d06630de53847515c9) commit).

We achieve that by only merging the subset of the original `lock_content` that contains those platforms.

Reply:
Yes, this seems reasonable to me.